### PR TITLE
DOC fix *_option() docstring

### LINKED ISSUE #28780: DOC: wrong formatting of the set_option docs
https://pandas-docs.github.io/pandas-docs-travis/reference/api/pandas.set_option.html

Notes section of the set_option docs seem to have lost some formatting/spacing on master.

PR coming to fix!

Ali

### PR DESCRIPTION
**Pull request by glemaitre**
- [x] closes #28780
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by glemaitre** [comment]
Is there any way to see the artefacts of the documentation built?

**Comment by glemaitre** [comment]
Do we want a non-regression test for this?

**Comment by alimcmaster1** [comment]
> Is there any way to see the artefacts of the documentation built?

Yes you can build the documentation locally. See https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#building-the-documentation

**Comment by alimcmaster1** [comment]
> Is there any way to see the artefacts of the documentation built?

Yes probably if not too much work

**Comment by glemaitre** [comment]
> Yes you can build the documentation locally. See https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#building-the-documentation

I was more referring through the CIs. I wanted to not build the documentation locally :)

**Comment by glemaitre** [comment]
> Yes probably if not too much work

I assume that you are referring to the unit test. I mentioning this because supposedly, you already have the numpydoc validation which is running (but not catching this one). It might be better to contribute upstream this case than just adding a single brittle test, isn't it?

**Comment by alimcmaster1** [comment]
> > Yes you can build the documentation locally. See https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#building-the-documentation
> 
> I was more referring to the artefacts stored by the CIs. I wanted to not build the documentation locally :)

Gotcha yeah sadly we don't (to my knowledge) create any docs artefacts in azure for PRs.
We easily could though: https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/pipeline-artifacts?view=azure-devops&tabs=yaml. I usually just build locally and show the output in my PR :)

File: pandas/_config/config.py [hunk 1]
```diff
@@ -158,6 +158,9 @@ def _describe_option(pat: str = "", _print_desc: bool = True):
     s = ""
     for k in keys:  # filter by pat
         s += _build_option_description(k)
+        s += "\n"
+    # strip the last newline
+    s = s[: s.rfind("\n")]
```
Comment (jorisvandenbossche) [inline pandas/_config/config.py]: Is this last newline a problem?

Comment (glemaitre) [reply pandas/_config/config.py]: Solving the issue as you mentioned by joining the description.

**Comment by glemaitre** [comment]
I build locally for the `pd.*_option` and it solves the issue (kudos for being able to build a single function :))

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**